internal/statesync/reactor.go

```diff
@@ -831,20 +835,41 @@ func (r *Reactor) handleMessage(ctx context.Context, chID p2p.ChannelID, envelop
 // encountered during message execution will result in a PeerError being sent on
 // the respective channel. When the reactor is stopped, we will catch the signal
 // and close the p2p Channel gracefully.
-func (r *Reactor) processCh(ctx context.Context, ch *p2p.Channel, chName string) {
-	iter := ch.Receive(ctx)
+func (r *Reactor) processChannels(ctx context.Context, chs ...*p2p.Channel) {
+	// make sure that the iterator get cleaned up correctly in the
+	// case of an error
```

Comment:
This is really just a number-agreement nit. 🙂 
```suggestion
	// make sure that the iterator gets cleaned up in case of error
```